### DISCUSSION TITLE
Word abstraction in utterances

### DISCUSSION THREAD

**Discussion by eneadodi** [aurelio-labs/semantic-router#171] [Ideas]
I got a quick question about the semantic router.
Is there a feature where you pass in utterances with a 'generic' word in such a way that the Route vector spaces consider that 'generic' word as variable?

For example, if we are working with soccer teams rather than having utterances:

"What players are in Barcelona?"
"Who is the owner of Roma?"
etc etc

you do :
"What players are in {X}?"
"Who is the owner of {X}?"

this would help in the classificaiton of new queries as it would ignore the team name entirely and consider it variable.

.train(X,y) helps a little bit with this but I don't think it is a permant solution.

Thanks,
Enea Dodi

**Comment by bruvduroiu** [comment]
Great question. I've been running into similar problems. Currently we don't handle generic words, and the suggested method of running should be to include as many relevant examples in the utterances parameter.

I'm not aware of any masking mechanism we could use with current embedding models that would allow us to use generic statements like `Who is the owner of {X}`.

You could also try including a substring in the utterance,

e.g.

```
"What players are in", 
"Who is the owner of"
```
